src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,103 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] map;
+
+    public MyHashMap() {
+        map = new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
+
+    private static class Node<K,V> {
+        private final K key;
+        private V value;
+        private MyHashMap.Node<K, V> next;
+
+        public Node(K key, V value, MyHashMap.Node<K,V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int hash = getHash(key);
+        int index = getIndex(hash);
+        if (size == 0) {
+            map[index] = new Node<>(key, value, null);
+            size++;
+        } else {
+            Node<K, V> node = map[index];
+            if (node == null) {
+                map[index] = new Node<>(key, value, null);
+                size++;
```

Comment:
you can move this code into a separate method, then your code will look more compact